### ISSUE TITLE
Is there the way to manage logs (request/response) via User Interface on X-Road?

### ISSUE THREAD

**Issue by ngolongquan** [nordic-institute/X-Road#363]
Hi @petkivim ,
Currently , I check the log files on server and query directly from table [logrecord] on [messagelog] database to mange request/response on X-Road .

I want to manage the logs (request/response) via User Interface on X-Road,
So, can you instruct me to the manage them via User Interface on X-Road?

Best Regards,
Ngo Long Quan

**Comment by petkivim** [comment]
Hi @ngolongquan 

Message logs are not accessible through the Security Server UI, but there's an API to download log records. In addition, the Security Server provides a tool for verifying the signatures of downloaded  messages. More information about the API and the verification tool can be found at:

https://github.com/nordic-institute/X-Road/blob/develop/doc/Manuals/ug-sigdoc_x-road_signed_document_download_and_verification_manual.md

Best regards,
Petteri

**Comment by ngolongquan** [comment]
Hi @petkivim ,
Thanks for your answer.
I'm searching with following link: 
https://github.com/ria-ee/X-Road-opmonitor/blob/master/docs/opendata/user_guide/ug_opendata_interface.md
https://github.com/ria-ee/X-Road-opmonitor/blob/master/docs/opendata/anonymizer.md

So, can i install it into X-Road 6.22.1?
And what's the package can i install?

Best Regards,
Ngo Long Quan

**Comment by petkivim** [comment]
Hi @ngolongquan 

Operational monitoring is an X-Road extension, and it is not part of the X-Road core. Currently there are no installation packages available. Instead, please follow the installation and configuration process described here:

https://github.com/ria-ee/X-Road-opmonitor#installation-instructions

Operational monitoring is currently maintained by the Estonian Information System Authority (RIA), not by NIIS. In case you need support regarding the installation and/or configuration, please contact RIA's support directly.

Best regards,
Petteri

**Comment by ngolongquan** [comment]
Hi @petkivim ,
Thanks for your support.
I got it, I will try it.

Best Regards,
Ngo Long Quan

**Comment by ngolongquan** [comment]
Hi @petkivim,
I have installed xroad-opmonitor addon - X-Road opmonitor daemon. However, I don't know how to it work. I made request to security server and received response (request/response) via REST.
The table [operational_data] in database [op-monitor] still is empty.

![opmonitorIssue](https://user-images.githubusercontent.com/60766273/74326809-a3c55e00-4dbd-11ea-9381-a8fa2bc16e71.png)

So, when is the table [operational_data] contains data?
Can you help me to clear?

Best Regards,
Ngo Long Quan

**Comment by petkivim** [comment]
Hi @ngolongquan 

How's the situation now?

Due to the asynchronous nature of the operational monitoring system, the overhead of request exchange is minimal, when operational data is stored. Operational data about each request is written to an in-memory operational data buffer at the security server and forwarded to the operational monitoring daemon in an asynchronous manner. Therefore, there's a delay before the data is present in thedatabase.

Best regards,
Petteri

**Comment by ngolongquan** [comment]
Hi @petkivim ,
Thanks for your answer.
Today, I re-check the database, it's still empty.
I don't know why.

Best Regards,
Ngo Long Quan

**Comment by ngolongquan** [comment]
Hi @petkivim,
I checked the /var/log/xroad/monitor.log then it log the following error:

2020-02-13 07:11:49,525 ERROR [xroad-monitor-akka.actor.default-dispatcher-2] akka.remote.EndpointWriter - AssociationError [akka.tcp://xroad-monitor@127.0.0.1:2552] -> [akka.tcp://Proxy@127.0.0.1:5567]: Error [Shut down address: akka.tcp://Proxy@127.0.0.1:5567] [
akka.remote.ShutDownAssociation: Shut down address: akka.tcp://Proxy@127.0.0.1:5567
Caused by: akka.remote.transport.Transport$InvalidAssociationException: The remote system terminated the association because it is shutting down.
]

Is this the root cause?

Best Regards,
Ngo Long Quan

**Comment by petkivim** [comment]
Hi  @ngolongquan 

Yes, that's probably the root cause. What's the host operating system and its version? In addition to the Security Server version 6.22.1, are you running any other components on the same host? The operational monitoring central component (https://github.com/ria-ee/X-Road-opmonitor) must not be run on the same host with the Security Server.

Best regards,
Petteri

**Comment by ngolongquan** [comment]
Hi @petkivim ,
I've installed xroad-opmonitor addon to the Security Server :(
And my Security Server is at version 6.22.1.
Thanks for your answer.
I'll try install xroad-opmonitor addon to the other server.

Best Regards,
Ngo Long Quan

**Comment by petkivim** [comment]
Hi @ngolongquan 

How about the host operating system and its version? Are you using Ubuntu 18.04 or Red Hat Enterprise Linux 7 (RHEL7)?

`xroad-opmonitor `package should be installed on the Security Server. However, the operational monitoring central component (https://github.com/ria-ee/X-Road-opmonitor) must be installed on a separate host - not on the Security Server.

Best regards,
Petteri

**Comment by ngolongquan** [comment]
Hi @petkivim,
The Security Server hosted OS: Ubuntu 18.04. And i install X-Road package from: https://github.com/nordic-institute/X-Road/blob/master/doc/Manuals/ig-ss_x-road_v6_security_server_installation_guide.md

> xroad-opmonitor package should be installed on the Security Server. However, the operational monitoring central component (https://github.com/ria-ee/X-Road-opmonitor) must be installed on a separate host - not on the Security Server.

It means i must install xroad-opmonitor package on Security Server and install the operational monitoring central component to the other (not on the Security Server). Is that right?

The reference link is maintained by ria-ee: (https://github.com/ria-ee/X-Road-opmonitor)
So, the xroad-opmonitor package support by Nordic Institute, right?



Best Regards,
Ngo Long Quan

**Comment by petkivim** [comment]
Hi @ngolongquan 

> It means i must install xroad-opmonitor package on Security Server and install the operational monitoring central component to the other (not on the Security Server). Is that right?

Yes, this is correct.

> So, the xroad-opmonitor package support by Nordic Institute, right?

Yes, the `xroad-opmonitor` package is an official Security Server addon, and it is maintained by NIIS.

>  The reference link is maintained by ria-ee: (https://github.com/ria-ee/X-Road-opmonitor)

This is a separate component that is not part of the X-Road core,  and it is maintained by RIA.

Best regards,
Petteri

**Comment by ngolongquan** [comment]
Hi @petkivim ,
I got it, thanks for your explanation.

> In addition, to be able to collect operational monitoring data from Security Servers, you must install the xroad-opmonitor addon which is available as an X-Road package in the official X-Road release repository.
> 
> https://github.com/nordic-institute/X-Road/blob/develop/doc/OperationalMonitoring/Architecture/arc-opmond_x-road_operational_monitoring_daemon_architecture_Y-1096-1.md 

> I checked the /var/log/xroad/monitor.log and it logs the following error:
> 
> 2020-02-13 07:11:49,525 ERROR [xroad-monitor-akka.actor.default-dispatcher-2] akka.remote.EndpointWriter - AssociationError [akka.tcp://xroad-monitor@127.0.0.1:2552] -> [akka.tcp://Proxy@127.0.0.1:5567]: Error [Shut down address: akka.tcp://Proxy@127.0.0.1:5567] [
> akka.remote.ShutDownAssociation: Shut down address: akka.tcp://Proxy@127.0.0.1:5567
> Caused by: akka.remote.transport.Transport$InvalidAssociationException: The remote system terminated the association because it is shutting down.

]

> Yes, that's probably the root cause. What's the host operating system and its version? In addition to the Security Server version 6.22.1, are you running any other components on the same host? The operational monitoring central component (https://github.com/ria-ee/X-Road-opmonitor) must not be run on the same host with the Security Server.

--> My Security Server (version 6.22.1) hosted on Ubuntu 18.04, and do not have any component on same host (just install xroad-opmonitor addon),  and separate PostgreSQL to the other Server.

My old issue: I'm not able to collect operational monitoring data from Security Servers. :(

Best Regards,
Ngo Long Quan

**Comment by ngolongquan** [comment]
Hi @petkivim,
I've checked the op-monitor.log, the snippet content following:

OpMonitorDaemon-akka.actor.default-dispatcher-2] INFO  org.hibernate.Version - HHH000412: Hibernate Core {[WORKING]}
2020-02-13 12:00:00,304 [OpMonitorDaemon-akka.actor.default-dispatcher-2] INFO  org.hibernate.cfg.Environment - HHH000206: hibernate.properties not found
2020-02-13 12:00:01,148 [OpMonitorDaemon-akka.actor.default-dispatcher-2] INFO  o.h.annotations.common.Version - HCANN000001: Hibernate Commons Annotations {5.0.4.Final}
2020-02-13 12:00:02,557 [OpMonitorDaemon-akka.actor.default-dispatcher-2] INFO  o.h.e.j.c.i.ConnectionProviderInitiator - HHH000130: Instantiating explicit connection provider: org.hibernate.hikaricp.internal.HikariCPConnectionProvider
2020-02-13 12:00:02,579 [OpMonitorDaemon-akka.actor.default-dispatcher-2] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Started.
2020-02-13 12:00:02,797 [OpMonitorDaemon-akka.actor.default-dispatcher-2] INFO  org.hibernate.dialect.Dialect - HHH000400: Using dialect: ee.ria.xroad.common.db.CustomPostgreSQLDialect
**2020-02-13 12:00:03,048 [OpMonitorDaemon-akka.actor.default-dispatcher-2] INFO  o.h.e.j.e.i.LobCreatorBuilderImpl - HHH000424: Disabling contextual LOB creation as createClob() method threw error : java.lang.reflect.InvocationTargetException**
2020-02-13 12:00:03,064 [OpMonitorDaemon-akka.actor.default-dispatcher-2] INFO  org.hibernate.type.BasicTypeRegistry - HHH000270: Type registration [java.util.UUID] overrides previous : org.hibernate.type.UUIDBinaryType@37ef1f05
2020-02-13 12:00:03,196 [OpMonitorDaemon-akka.actor.default-dispatcher-2] WARN  org.hibernate.orm.deprecation - HHH90000014: Found use of deprecated [org.hibernate.id.SequenceHiLoGenerator] sequence-based id generator; use org.hibernate.id.enhanced.SequenceStyleGenerator instead.  See Hibernate Domain Model Mapping Guide for details.
2020-02-13 12:00:04,044 [OpMonitorDaemon-akka.actor.default-dispatcher-2] INFO  o.h.h.i.QueryTranslatorFactoryInitiator - HHH000397: Using ASTQueryTranslatorFactory
2020-02-13 12:00:04,185 [OpMonitorDaemon-akka.actor.default-dispatcher-2] INFO  e.r.x.o.OperationalDataRecordCleaner - No outdated operational data records to remove from the database

Does the bold text relate this issue?

Best Regards,
Ngo Long Quan
